geoportailv3/static/js/draw/drawdirective.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @fileoverview This file provides a draw directive. This directive is used
+ * to create a draw panel in the page.
+ *
+ * Example:
+ *
+ * <app-draw app-draw-map="::mainCtrl.map"></app-draw>
+ *
+ * Note the use of the one-time binding operator (::) in the map expression.
+ * One-time binding is used because we know the map is not going to change
+ * during the lifetime of the application.
+ */
+goog.provide('app.DrawController');
+goog.provide('app.drawDirective');
+
+goog.require('app');
+goog.require('ol.events.condition');
+goog.require('ol.interaction.Draw');
+goog.require('ol.interaction.Select');
+
+
+/**
+ * @param {string} appDrawTemplateUrl Url to draw template
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ */
+app.drawDirective = function(appDrawTemplateUrl) {
+  return {
+    restrict: 'E',
+    scope: {
+      'map': '=appDrawMap',
+      'features': '=appDrawFeatures',
+      'active': '=appDrawActive'
+    },
+    controller: 'AppDrawController',
+    controllerAs: 'ctrl',
+    bindToController: true,
+    templateUrl: appDrawTemplateUrl
+  };
+};
+
+
+app.module.directive('appDraw', app.drawDirective);
+
+
+
+/**
+ * @param {!angular.Scope} $scope Scope.
+ * @param {ngeo.DecorateInteraction} ngeoDecorateInteraction Decorate
+ *     interaction service.
+ * @constructor
+ * @export
+ * @ngInject
+ */
+app.DrawController = function($scope, ngeoDecorateInteraction) {
+
+  /**
+   * @type {ol.Map}
+   *  @export
+   */
+  this.map;
+
+  /**
+   * @type {boolean}
+   *  @export
+   */
+  this.active;
+
+  /**
+   * @type {number}
+   * @private
+   */
+  this.featureSeq_ = 0;
+
+  /**
+   * @type {ol.Collection.<ol.Feature>}
+   * @export
+   */
+  this.features;
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  var drawPoint = new ol.interaction.Draw({
+    features: this.features,
+    type: /** @type {ol.geom.GeometryType} */ ('Point')
+  });
+  /**
+   * @type {ol.interaction.Draw}
+   * @export
+   */
+  this.drawPoint = drawPoint;
+
+  drawPoint.setActive(false);
+  ngeoDecorateInteraction(drawPoint);
+  this.map.addInteraction(drawPoint);
+  goog.events.listen(drawPoint, ol.interaction.DrawEventType.DRAWEND,
+      angular.bind(this, this.onDrawEnd_, drawPoint));
```

Comment:
Don't use `angular.bind` in compiled code. And you don't need to use `bind` at all here. Instead you can just use the following:

```js
  goog.events.listen(drawPoint, ol.interaction.DrawEventType.DRAWEND,
      this.onDrawEnd_, false, this);
```

and use `event.target` in `onDrawEnd_` to get a reference to the interaction to which the `drawend` event was dispatched.